source/common/common/utility.h

```diff
@@ -551,7 +552,38 @@ template <class Value> struct TrieEntry {
  * A trie used for faster lookup with lookup time at most equal to the size of the key.
  */
 template <class Value> struct TrieLookupTable {
-
+  ~TrieLookupTable() {
+    // Use a stack to store non-nullptr nodes which should be deleted, and their children will be
+    // pushed into this stack and will be deleted later.
+    std::stack<TrieEntry<Value>*> to_delete;
+    // Release all pointers in root_ and push them into the stack.
+    for (unsigned long i = 0; i < root_.entries_.size(); i++) {
+      auto entry = std::move(root_.entries_[i]); // This entry in trie will be nullptr now.
+      if (entry != nullptr) {
+        to_delete.push(
+            entry.release()); // Raw points will be inside the stack and will be deleted later.
+      }
+    }
+    // Now all unique_ptr in root_ are released and raw pointers are stored inside the stack.
+    // We will delete all of the raw pointers in below while loop.
+    // All raw pointers in to_delete will finally be deleted and popped out.
+    while (!to_delete.empty()) {
+      TrieEntry<Value>* current = to_delete.top();
+      to_delete.pop();
+      bool empty_entry = true;
```

Comment:
I don't see where `empty_entry` is used.  Can it be removed?

Reply:
> I don't see where `empty_entry` is used. Can it be removed?

Sorry I forgot to remove it. Fixed!